### PR TITLE
feat(filter): add support to filter array data based on wildcard `*`

### PR DESCRIPTION
**Pull request by tujoworker**

### CODE REVIEW

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Handler/info.mdx [hunk 1]
```diff
@@ -142,4 +142,8 @@ It returns the filtered form data.
 
 The [useData](/uilib/extensions/forms/Form/useData/#filter-data) hook and the [getData](/uilib/extensions/forms/Form/getData/#filter-data) method also returns a `filterData` function you can use to filter data the same way.
 
-Down below is an example of how to use the `filterSubmitData` method.
+The the demo section is an example of how to use the `filterSubmitData` method.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Form/Handler/info.mdx:145]:
```suggestion
The demo section is an example of how to use the `filterSubmitData` method.
```

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/info.mdx [hunk 2]
```diff
@@ -73,3 +73,42 @@ render(
...
+## Filter data
+
+You can filter data by paths specific or all paths.
+
+- `/myList/0` will filter out the first item of the list, including `foo` and `bar`.
+- `/myList/1/foo` will filter out `foo` inside the second item of the list.
+- `/myList/*/foo` will filter out all `foo` object keys from all items of the list.
+- `/myList/*/subList/*/foo` does support multi wildcard paths.
+
+In the example above, the data given in `onSubmit` will still have "bar2" and "bar2" of the list.
```
Comment (langz) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/info.mdx:86]: I have not read this to great detail yet, but is it in the example above or below? And is «bar2» and «bar2» correct?

**Review by langz — APPROVED** [review]

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/info.mdx (hunk 2 above)
Comment (tujoworker) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/Iterate/Array/info.mdx:86]: Yes, both where wrong 🙏

**Pull request merged**